main.go

```diff
@@ -455,3 +473,69 @@ var outputHeader = lipgloss.NewStyle().Foreground(lipgloss.Color("#F1F1F1")).Bac
 func printFilenameOutput(filename string) {
 	fmt.Println(lipgloss.JoinHorizontal(lipgloss.Center, outputHeader.String(), filename))
 }
+
+type Positions struct {
+	Left   float64
+	Center float64
+	Right  float64
+	Top    float64
+}
+
+func getPositions(config Config, x, y, imgW, controlsWidth float64) Positions {
+	return Positions{
+		Left:   config.Margin[left] + x + controlsWidth,
+		Center: imgW / 2,
+		Right:  imgW - (config.Margin[right] + x),
+		Top:    config.Margin[top] + y,
+	}
+}
+
+func (title Title) Validate(value string) error {
+	if value == "" || value == "-" {
+		return errors.New("Invalid title text provided.")
+	}
+	switch title.Position {
+	case "left", "center", "right":
+		return nil
+	default:
+		return errors.New("Invalid title position. Must be one of \"left\", \"center\", or \"right\".")
+	}
+}
+
+// NewWindowTitle returns a title element with the given text.
+func NewWindowTitle(config Config, positions Positions, scale float64, s *chroma.Style) (*etree.Element, error) {
+	titleText := config.Title.Text
```

Comment:
There is no need, the previous code was OK, you simply had to remove the value string from the function signature.

It's OK, and better, to pass nothing to validate, it validates the Title object